alchemist/alchemist-swingui/src/main/kotlin/it/unibo/alchemist/boundary/gui/view/properties/RangedDoubleProperty.kt

```diff
@@ -0,0 +1,95 @@
+package it.unibo.alchemist.boundary.gui.view.properties
+
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import com.google.gson.JsonSerializationContext
+import it.unibo.alchemist.boundary.gui.view.properties.PropertyTypeAdapter.NAME
+import it.unibo.alchemist.boundary.gui.view.properties.PropertyTypeAdapter.VALUE
+import javafx.beans.property.DoublePropertyBase
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+import java.io.Serializable
+import java.lang.reflect.Type
+
+private const val LOWER_BOUND = "lower bound"
+private const val UPPER_BOUND = "upper bound"
+
+open class RangedDoubleProperty @JvmOverloads constructor(
```

Comment:
Missing Kdoc, can this one be closed?

Reply:
Not checking for null in equals is always a problem. You're welcome to fix.

Reply:
I just noticed that the first method call is a safe call through the ?. operator, my bad.